uaa.go

```diff
@@ -72,15 +72,16 @@ func (c *UAAClient) CreateClient(client Client) (Client, error) {
 		return Client{}, err
 	}
 
-	if resp.StatusCode != 201 {
-		return Client{}, fmt.Errorf("Expected status 201; got: %d", resp.StatusCode)
-	}
-
-	err = decodeBody(resp, &client)
+	output := map[string]any{}
+	err = decodeBody(resp.Body, &output)
 	if err != nil {
 		return Client{}, err
 	}
 
+	if resp.StatusCode != 201 {
+		return Client{}, fmt.Errorf("expected status 201; got: %d. error: %s", resp.StatusCode, output)
+	}
+
 	return client, nil
```

Comment:
It looks like you're still returning `client` but you no longer decode the body into it.